### ISSUE TITLE
혹시 이전 버전을 공유해주실 수 있으신가요?

### ISSUE THREAD

**Issue by yoonespi** [gusdnd852/kochat#4]
안녕하세요, 자연어 처리를 공부해보려고 합니다.
의도 파악을 위한 튜토리얼을 찾다가 좋은 자료가 있다고 해서 찾아왔는데,
현재 업데이트 중이셔서 코드가 돌아가지 않는다고 보았습니다.

혹시 가능하시다면 이전 버전에 대한 코드를 공유해줄 수 있으신가요?

**Comment by gusdnd852** [comment]
커밋버튼 누르시면 예전 버전코드들 보실 수는 있긴한데, 그 코드를 제가 완전히 혼자 짠게 아니고 여기저기서 긁어온 코드들 돌아가게만 붙인거라 코드들 간에 일관성이나 가독성도 심하게 떨어지고, 좋은 코드가 아닙니다. (당시에 학교에서 과제마감 얼마 안남았을 때 시작해서 급하게 제출하느라 그런거 신경쓸 겨를도 없었네요 ㅎㅎ) 그래서 지금 OOP 적용해서 아예 처음부터 디자인해서 Pytorch로 구현하고 있습니다. 이전에는 모델도 한가지였는데 모델도 여러가지로 해서 고를 수 있고, 의도파악시에 발생할 수 있는 Fallback도 여러가지 구현방식으로 테스트 해보려고합니다.

의도파악 튜토리얼이라면, 현재까지 간단하게 구현된 코드만으로도 의도파악 부분은 학습-테스트까지는 가능하긴 한데요. 이전 코드도 그렇고 지금 간단하게 구현된 버전도 Softmax Classification한거라 Softmax Score값으로 Fallback 체크해야하는데, 이게 좋은 방식이 아닙니다. [요런 논문들](https://arxiv.org/abs/1412.1897) 보시면 Softmax Score가 제대로  Calibration되지 않았는데 그걸 그대로 Probability라고 이해하고 사용하는 경우에 어떤 문제가 발생할 수 있는지 여실히 드러납니다. (이전버전도 이렇게 구현한거라 좋은 코드가 아닙니다.) 이 문제를 해결하기 위해서 이번주 주말부터 해서 각 의도의 인트라클래스간에 마진을 줘서 Distance 기반으로 Retrieval하는 방식으로 Fallback Checking하는 코드를 구현중인데, 아마 금방 끝날 것 같습니다. (아마 2주안에 챗봇에 필요한 Intent Detection이랑 NER은 모두 구현되지 않을까 싶습니다) 예전 버전 코드 보시는 것보다 2주뒤쯤에 다시 확인하시는게 좋아보입니다.. (당장에 예전 코드 보려면 Commits 버튼 눌러서 거의 맨 처음아니면 거의 극초반 버전 모시면 되긴 합니다만... 비추합니다. 부끄럽네요 ㅎㅎ)

**Comment by yoonespi** [comment]
아 그렇군요 감사합니다! commit을 찾아보겠습니다.
2주 뒤 멋진 알고리즘을 기대하겠습니다!!

**Comment by gusdnd852** [comment]
업데이트가 모두 완료되었습니다. 확인 바랍니다.

**Comment by yoonespi** [comment]
> @yoonespi
> 업데이트가 모두 완료되었습니다. 확인 바랍니다.

친절히 알려주셔서 감사합니다!
찬찬히 읽어볼게요
